scality_manila_utils/cli.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2015 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import grp
+import logging
+import logging.handlers
+import os
+import pwd
+
+from scality_manila_utils.helper import Helper
+from scality_manila_utils.exceptions import EnvironmentException
+
+log = logging.getLogger(__name__)
+
+
+def setup_logger():
+    """
+    Setup root log handler.
+    """
+    log_format = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+    )
+    handler = logging.handlers.SysLogHandler('/dev/log')
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(log_format)
+    logging.root.addHandler(handler)
+    logging.root.setLevel(logging.INFO)
+
+
+def drop_privileges():
+    """
+    Attempt to drop privileges.
+    """
+    def _try_get(f, *args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except KeyError:
+            return None
+
+    def get_user(name):
+        return _try_get(pwd.getpwnam, name)
+
+    def get_group(name):
+        return _try_get(grp.getgrnam, name)
+
+    # Attempt to find a proper user and group
+    user_info = get_user('nobody')
+    candidate_groups = ('nogroup', 'nobody')
+    for group in candidate_groups:
+        group_info = get_group(group)
+        if group_info is not None:
+            break
+
+    # Drop privileges
+    if user_info and group_info:
+        log.debug('Dropping privileges to %s:%s', user_info.pw_name,
+                  group_info.gr_name)
+        os.setegid(group_info.gr_gid)
+        os.seteuid(user_info.pw_uid)
+    else:
+        msg = 'Unable to find an unprivileged user/group'
+        log.error(msg)
+        raise RuntimeError(msg)
+
+
+def main(args=None):
+    setup_logger()
+
+    if os.geteuid() != 0:
+        log.error('Invoked without superuser privileges')
+        raise RuntimeError("This program requires superuser privileges")
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
```

Comment:
Add a `--version` thingy so the Manila driver can query the version of `scality-manila-utils` installed on a remote system, and validate it's a supported version during it's system/configuration checks.